mantidproject/mantid
# Load files with periods with no data

It's possible for a user to set the number of periods in a run independently of how many periods are actually in the data. One case is where the number of periods specified in the file is greater than the number in the data, i.e. there are periods with no data. This fix will now let you load the file in this case, and will give you a warning message about it. If the number of periods specified is less than the number in the data, then the file will still refuse to load.

#### Summary of work

- Added logic for case of nperiods > number of periods in data, including warning message
- Added test to check the problem file in #33729 loads

Fixes #33729. 




### To test:
The file in the referenced issue (`LARMOR00062766.nxs`) should load into one workspace. `LARMOR00003368.nxs` is an example of a file with multiple periods (all with data), it's the one used in several of the existing tests. Other event files should still load okay.







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
